scripts/deployment-helpers.sh

```
@@ -1,14 +1,61 @@
+#   If MAAS_API_IMAGE is not set, does nothing (uses default from core/kustomization.yaml).
+#   Creates a backup and restores it after the calling script exits.
+#
+# Environment:
+#   MAAS_API_IMAGE - Container image to use (e.g., quay.io/opendatahub/maas-api:pr-123)
+set_maas_api_image() {
+  # Skip if MAAS_API_IMAGE is not set
+  if [ -z "${MAAS_API_IMAGE:-}" ]; then
+    return 0
+  fi
+
+  local project_root="$(find_project_root)"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)